### PR TITLE
Add LayoutMap which will be used for model parallelism

### PR DESCRIPTION
**Pull request by qlzh727**
1. The LayoutMap follows the existing keras.dtensor implementation, which is a dict like class that maps string to layout.
2. Added docstring for how to use the layout map.
3. Added unit test.
4. Model parallel distribution will be in a follow up cl.

### CODE REVIEW

File: keras_core/distribution/distribution_lib.py [hunk 1]
```diff
@@ -288,6 +289,102 @@ def get_variable_layout(self, variable):
...
+
+    See below for a usage example. You can define the naming schema
+    of the `TensorLayout`, and then retrieve the corresponding
+    `TensorLayout` instance.
+
+    In the normal case, the key to query is usually the `variable.path`, which
+    is the idenifier of the variable.
+
+    ```python
+    map = LayoutMap(device_mesh=None)
```
Comment (fchollet) [inline keras_core/distribution/distribution_lib.py:308]: `map` is a reserved keyword, so let's use something like `variable_map`

File: keras_core/distribution/distribution_lib.py [hunk 2]
```diff
@@ -288,6 +289,102 @@ def get_variable_layout(self, variable):
...
+    layout_3 = map['dense_2.kernel']                #   layout_3 == layout_2d
+    layout_4 = map['dense_2.bias']                  #   layout_4 == layout_1d
+    layout_5 = map['my_model/conv2d_123/kernel']    #   layout_5 == layout_4d
+    layout_6 = map['my_model/conv2d_123/bias']      #   layout_6 == layout_1d
+    layout_7 = map['my_model/conv3d_1/kernel']      #   layout_7 == None
+    layout_8 = map['my_model/conv3d_1/bias']        #   layout_8 == None
+    ```
+
+    Args:
+      device_mesh: An optional `DeviceMesh` that can be used to populate the
```
Comment (fchollet) [inline keras_core/distribution/distribution_lib.py:325]: Use 4-space indent

File: keras_core/distribution/distribution_lib.py [hunk 3]
```diff
@@ -288,6 +289,102 @@ def get_variable_layout(self, variable):
...
+    def __getitem__(self, key):
+        """Retrieve the corresponding layout by the string key.
+
+        When there isn't an exact match, all the existing keys in the layout map
+        will be treated as a regex and map against the input key again. The
+        first match will be returned, based on the key insertion order. Return
+        None if there isn't any match found.
+
+        Args:
+          key: the string key as the query for the layout.
```
Comment (fchollet) [inline keras_core/distribution/distribution_lib.py:342]: Same in other docstrings

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras_core/distribution/distribution_lib.py (hunk 1 above)
Comment (qlzh727) [reply keras_core/distribution/distribution_lib.py]: Done.

File: keras_core/distribution/distribution_lib.py (hunk 2 above)
Comment (qlzh727) [reply keras_core/distribution/distribution_lib.py]: Done.

File: keras_core/distribution/distribution_lib.py (hunk 3 above)
Comment (qlzh727) [reply keras_core/distribution/distribution_lib.py]: Done.

**Review by fchollet — APPROVED** [review]
LGTM, thank you

**Pull request merged**